go.go

```diff
@@ -259,6 +259,30 @@ const goMarshalField = `{{if eq .Type "bool"}}
 		i++
 	}
 {{else if eq .Type "int32"}}
+{{- if .TypeList}}
+	if l := len(o.{{.NameTitle}}); l != 0 {
+		buf[i] = {{.Index}}
+		i++
+		x := uint(l)
+		for x >= 0x80 {
+			buf[i] = byte(x | 0x80)
+			x >>= 7
+			i++
+		}
+		buf[i] = byte(x)
+		i++
+		for _, v := range o.{{.NameTitle}} {
+			x1 := uint32((v << 1) ^ (v >> 31))
```

Comment:
You can use the CPU's natural size as in `uint(v << 1) ^ uint(v >> 31)`.

Reply:
Dito for the marshal length function below.

Reply:
Then negative numbers take up more space
For example, -1690982682